### PR TITLE
docs: add sample for timeout for one RPC

### PR DESCRIPTION
**Pull request by olavloite**
Adds a sample for setting a timeout for a single RPC.

Fixes https://github.com/GoogleCloudPlatform/java-docs-samples/issues/3805

### CODE REVIEW

**Comment by eaball35** [comment]
Screen Shot 2020-12-10 at 9 00 28 AM

**Comment by olavloite** [comment]
@eaball35 These `RESOURCE_EXHAUSTED` errors have been mitigated by https://github.com/googleapis/java-spanner/pull/669, which is included in version 3.1.0. The sample builds are currently using version 3.0.4. I'm afraid the only possible mitigation at this moment is retrying the test run.

**Review by thiagotnunes — APPROVED** [review]

File: samples/snippets/src/main/java/com/example/spanner/StatementTimeoutExample.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    Context context =
+        Context.current().withValue(SpannerOptions.CALL_CONTEXT_CONFIGURATOR_KEY, configurator);
+    // Run the transaction in the custom context.
+    context.run(new Runnable() {
+      public void run() {
+        client.readWriteTransaction().run(new TransactionCallable<long[]>() {
+          public long[] run(TransactionContext transaction) throws Exception {
+            String sql = "INSERT Singers (SingerId, FirstName, LastName)\n"
+                + "VALUES (20, 'George', 'Washington')";
+            long rowCount = transaction.executeUpdate(Statement.of(sql));
```
Comment (lesv) [inline samples/snippets/src/main/java/com/example/spanner/StatementTimeoutExample.java:74]: How does this work in a multiprocessing environment when several processes are trying to test this at the same time?

**Review by lesv — CHANGES_REQUESTED** [review]
Testing can't possibly ever work.

File: samples/snippets/src/main/java/com/example/spanner/StatementTimeoutExample.java (hunk 1 above)
Comment (olavloite) [reply samples/snippets/src/main/java/com/example/spanner/StatementTimeoutExample.java:74]: Each test run always creates a database that is only used for that test run:

https://github.com/googleapis/java-spanner/blob/f794dd43665b618e51002204b651049ea5aa013b/samples/snippets/src/test/java/com/example/spanner/SpannerStandaloneExamplesIT.java#L80-L95

Comment (eaball35) [reply samples/snippets/src/main/java/com/example/spanner/StatementTimeoutExample.java:74]: Should work to create and drop db before and after each test, but dbs need be unique so they don't overlap. Will dbs be unique with current use of instanceid/databseId?

Comment (olavloite) [reply samples/snippets/src/main/java/com/example/spanner/StatementTimeoutExample.java:74]: Yes, the `databaseId` is uniquely generated for each test run by taking the system property as a base id and appending a UUID:
1. Getting the base `databaseId`: https://github.com/googleapis/java-spanner/blob/e6a34d924dff94e2f238eec63b9e62366d3cb506/samples/snippets/src/test/java/com/example/spanner/SpannerStandaloneExamplesIT.java#L52
2. Appending a UUID: https://github.com/googleapis/java-spanner/blob/2b3775a02466176695d7b88312b17c1aeedfbc16/samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java#L439-L441

**Review by eaball35 — APPROVED** [review]

**Comment by olavloite** [comment]
@lesv Would you mind taking a second look at this PR? The PR is currently blocked from merging as you requested changes.

**Comment by lesv** [comment]
@olavloite  Apologies for the delay, what I'm concerned about is that we might test with many PR's / java version tests running at the same time, I'm concerned that you will have collisons.

I was probably stuck on table names of Singers & Venues, but that's not really the problem since you are creating an instance and then deleting it.  The code in `createTestDatabase` appears to use EnvVar, it's unclear if that's unique.  It also falls back to using the last instanceId, but that looks like it should at least have a while in there somewhere, and actually go to a next id rather than the last one to have something unique.  

I'm going to release my objection so you can move forward - again I missed the email or I'd have responded earlier.  Apologies.

If you believe I'm wrong in my concern, feel free to merge, if not could you fix it and explain where my understanding is flawed?

**Review by lesv — APPROVED** [review]

**Comment by lesv** [comment]
I found the discussion. LGTM

**Pull request merged**